src/prommis/nanofiltration/diafiltration_two_salt.py

```diff
@@ -692,16 +747,31 @@ def _lumped_water_flux(self, x):
         self.lumped_water_flux = Constraint(self.x_bar, rule=_lumped_water_flux)
 
         def _D_lithium_lithium_calculation(self, x, z):
-            params_lithium_lithium = [-3.79e-06, -3.64e-08, 7e-09]
+            if value(self.membrane_fixed_charge) == 0:
+                # these params assume chi=0 mM and lithium & cobalt mem_conc range 50-80 mM
+                self.diffusion_params_lithium_lithium = [-4.07e-06, -3.99e-09, 4.01e-09]
+            if value(self.membrane_fixed_charge) == -140:
```

Comment:
I always get nervous when using exact equality with floating point numbers. At the very least, you should have an `else` clause for the case when `self.membrane_fixed_charge != 0 and self.membrane_fixed_charge != -140`.

Also, you shouldn't be creating a block attribute inside a rule for generating a `Constraint` or `Expression`. Furthermore, you shouldn't be using `self` as a function argument because it obscures what's actually happening inside the function---you're creating a new variable "self" that shadows the `self` keyword from the outer scope. Much better to use `b` or `blk` when defining a rule.